projects/workforce-allocations-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/allocations/AllocatePersonService.kt

```diff
@@ -0,0 +1,172 @@
+package uk.gov.justice.digital.hmpps.integrations.delius.allocations
+
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.audit.AuditedInteraction
+import uk.gov.justice.digital.hmpps.audit.service.AuditedInteractionService
+import uk.gov.justice.digital.hmpps.datetime.DeliusDateTimeFormatter
+import uk.gov.justice.digital.hmpps.exception.ConflictException
+import uk.gov.justice.digital.hmpps.exception.NotFoundException
+import uk.gov.justice.digital.hmpps.integrations.delius.audit.BusinessInteractionCode
+import uk.gov.justice.digital.hmpps.integrations.delius.contact.Contact
+import uk.gov.justice.digital.hmpps.integrations.delius.contact.ContactContext
+import uk.gov.justice.digital.hmpps.integrations.delius.contact.ContactRepository
+import uk.gov.justice.digital.hmpps.integrations.delius.contact.ContactTypeCode
+import uk.gov.justice.digital.hmpps.integrations.delius.contact.ContactTypeRepository
+import uk.gov.justice.digital.hmpps.integrations.delius.contact.findByCodeOrThrow
+import uk.gov.justice.digital.hmpps.integrations.delius.person.PersonManager
+import uk.gov.justice.digital.hmpps.integrations.delius.person.PersonManagerRepository
+import uk.gov.justice.digital.hmpps.integrations.delius.person.PersonRepository
+import uk.gov.justice.digital.hmpps.integrations.delius.person.ResponsibleOfficer
+import uk.gov.justice.digital.hmpps.integrations.delius.person.ResponsibleOfficerRepository
+import uk.gov.justice.digital.hmpps.integrations.workforceallocations.AllocationDetail.PersonAllocationDetail
+
+@Service
+class AllocatePersonService(
+    private val auditedInteractionService: AuditedInteractionService,
+    private val personRepository: PersonRepository,
+    private val personManagerRepository: PersonManagerRepository,
+    private val allocationValidator: AllocationValidator,
+    private val contactTypeRepository: ContactTypeRepository,
+    private val contactRepository: ContactRepository,
+    private val responsibleOfficerRepository: ResponsibleOfficerRepository,
+
+) : ManagerService<PersonManager>(personManagerRepository) {
+
+    @Transactional
+    fun createPersonAllocation(allocationDetail: PersonAllocationDetail) {
+        val person = personRepository.findByCrn(allocationDetail.crn)
+            ?: throw NotFoundException("Person", "crn", allocationDetail.crn)
+
+        auditedInteractionService.createAuditedInteraction(
+            BusinessInteractionCode.ADD_PERSON_ALLOCATION,
+            AuditedInteraction.Parameters("offenderId" to person.id.toString())
+        )
```

Comment:
Shouldn't we also be auditing with Outcome=FAIL if it fails to process later on?